pkg/node/client.go

```diff
@@ -193,6 +207,51 @@ func (sc *Client) getIngressNodes(ctx context.Context) ([]Node, error) {
 	return nodes, nil
 }
 
+// filterClientsByNodeGroups filters bee clients by node groups
+// Node names in beekeeper deployments follow the pattern: {nodeGroup}-{index}
+// This function extracts the node group from the node name and filters accordingly
+func (sc *Client) filterClientsByNodeGroups(clients map[string]*bee.Client) map[string]*bee.Client {
```

Comment:
You do not need this logic. Much simplier and more precise solution would be to add `NodeGroupName` in `bee.ClientOptions`, and to expose new method NodeGroup() in `bee` package. Then you can compare on name. This should be done only in case of beekeeper deployment.

Reply:
This is psudo code how this filtering could work:
```go
	filteredClients := make(map[string]*bee.Client)
	if len(sc.nodeGroups) > 0 && sc.deploymentType == DeploymentTypeBeekeeper {
		for _, ng := range sc.nodeGroups {
			for name, client := range sc.beeClients {
				if client.NodeGroup() == ng {
					filteredClients[name] = client
				}
			}
		}
	} else {
		filteredClients = sc.beeClients
	}
```